students/jimin/users/views.py

```diff
@@ -14,31 +14,59 @@ def post(self, request):
-             REGEX_PASWWORD   = '^(?=.*[A-Za-z])(?=.*\d)(?=.*[$@$!%*#?&])[A-Za-z\d$@$!%*#?&]{8,25}$'
+             REGEX_PASSWORD   = '^(?=.*[A-Za-z])(?=.*\d)(?=.*[$@$!%*#?&])[A-Za-z\d$@$!%*#?&]{8,25}$'
              REGEX_PHONE_NUMBER = '\d{3}-\d{4}-\d{4}'
+             REGEX_LAST_NAME='^[a-zA-Z가-힣]+$'
+             REGEX_FIRST_NAME='^[a-zA-Z가-힣]+$'
              
              if not re.match(REGEX_EMAIL,email):
                  return JsonResponse({'message' : 'INVALID_EMAIL'}, status = 400)
              
-             if not re.match(REGEX_PASWWORD,password):
+             if not re.match(REGEX_PASSWORD,password):
                  return JsonResponse({'message' : 'INVALID_PASSWORD'}, status = 400)
              
              if not re.match(REGEX_PHONE_NUMBER,phone_number):
                  return JsonResponse({'message' : 'INVALID_PHONE_NUMBER'}, status = 400)
              
              if User.objects.filter(email=email).exists():
-                 return JsonResponse({'message':'DUPLICATE_EMAIL'}, status=400)
-            
+                 return JsonResponse({'message': 'DUPLICATE_EMAIL'}, status=400)
+             
+             if not re.match(REGEX_LAST_NAME,last_name):
+                 return JsonResponse({'message' : 'INVALID_LAST_NAME'}, status = 400)
+             
+             if not re.match(REGEX_FIRST_NAME,first_name):
+                 return JsonResponse({'message' : 'INVALID_FIRST_NAME'}, status = 400)
+             
              User.objects.create(
-                first_name    = data['first_name'],
-                last_name     = data['last_name'],
-                email         = data['email'],
-                password      = data['password'],
-                phone_number  = data['phone_number'],
+                first_name   = data['first_name'],
+                last_name    = data['last_name'],
+                email        = data['email'],
+                password     = data['password'],
+                phone_number = data['phone_number'],
             )
              
              return JsonResponse({'message' : 'SUCCESS'}, status=201)
         except KeyError:
-             return JsonResponse({'message' : 'KEY_ERROR'}, status=400)
\ No newline at end of file
+             return JsonResponse({'message' : 'KEY_ERROR'}, status=400)
+
+class SignInView(View):
+    def post(self, request):
+        try :
+            data = json.loads(request.body)
+            
+            
+            if User.objects.filter(email=data['email']).exists():
+                user = User.objects.get(email = data['email'])
+            else :
+                return JsonResponse({'message':'NO_EMAIL'},status=400)
+            
+            if user.password == data['password']:
+                return JsonResponse({'message' : 'SUCCESS'}, status=201)
+            
+            return JsonResponse({'message' : 'INVALID_PASSWORD'}, status=201)
```

Comment:
- 로그인의 경우 아이디가 잘못 되었는지, 비밀번호가 잘못 되었는지 알 수 없게 메시지를 보내야 합니다.
- 인증과 관련된 상태코드는 401 입니다.
```suggestion
                return JsonResponse({'message':'INVALIDE_USER'},status=401)
```